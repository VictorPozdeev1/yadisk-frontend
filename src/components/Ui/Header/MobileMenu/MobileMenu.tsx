import React, { useState } from "react";
import { Drawer, Fab, useMediaQuery } from "@mui/material";
import AddIcon from "@mui/icons-material/Add";
import {Link, NavLink} from "react-router-dom";
import Spinner from "../../Spinner/Spinner";

interface MobileMenuProps {
    categories: any[];
    open: boolean;
    onClose: () => void;
}

const MobileMenu: React.FC<MobileMenuProps> = ({ categories, open, onClose }) => {
    const [selectedCategory, setSelectedCategory] = useState<string | null>(null);



    console.log(categories);
    const content = categories ? (

        categories.map((item: any) => (
            //<div key={item.resource_id}>{item.name}</div>
            <div
                key={item.resource_id}
                style={{
                    padding: "12px 24px 12px 24px",
                    width: "90%",
                    backgroundColor: selectedCategory === item.name ? "#ffffff" : "transparent",
                    borderRadius: selectedCategory === item.name ? "7px" : "transparent",
                    textAlign: "center",
                }}
            >
                <NavLink
                    key={item.resource_id}
                    to={`/${item.name}`}
                    style={{
                        textDecoration: "none",
                        color: selectedCategory === item.name ? "#035FA2" : "#333",
                    }}
                    onClick={() => setSelectedCategory(item.name)}
                >
                    {item.name}
                </NavLink>
            </div>
        ))
    ) : (
        <Spinner />
    );

    const isMobile = useMediaQuery('(max-width: 768px)');

    return (
        <Drawer anchor="top" open={open} onClose={onClose}  sx={{ height: "100%"}}>
            <div
                style={{
                    display: "flex",
                    flexDirection: "column",
                    alignItems: "center",
                    justifyContent: "center",
                    height: "100%",
                }}
            >
                <img
                    src="/images/logo.png"
                    alt="Росатом"
                    style={{ margin: "1em 1em",
                        width: '25%'
                    }}
                />


                <div style={{
                    padding: "12px 24px 12px 24px",
                    width: "90%",
                    backgroundColor: selectedCategory === null ? "#ffffff" : "transparent",
                    borderRadius: selectedCategory === null ? "7px" : "transparent",
                    textAlign: "center",
                }}>
                    <NavLink
                        key={"allDocuments"}
                        to={"/"}
                        onClick={() => setSelectedCategory(null)}
                        style={{
                            textDecoration: "none",
                            color: selectedCategory === null ? "#035FA2" : "#333",
                            cursor: "pointer",
                        }}
                    >
                        Все документы
                    </NavLink>
                </div>
                {content}

                <Fab
                    variant="extended"
                    sx={{
                        width: "30%",
                        backgroundColor: "#035FA2",
                        color: "white",
                        marginTop: "2em",
                        marginBottom: "2em",
                        display: "flex",
                        justifyContent: "space-between",
                        transition: "background-color 0.5s",
                        fontSize: "15px",
                        "&:hover": {
                            backgroundColor: "#024978",
                        },
                    }}
                    size="large"
                >
                    Документ
                    <AddIcon/>
                </Fab>
            </div>
        </Drawer>
    );
}

export default MobileMenu;
