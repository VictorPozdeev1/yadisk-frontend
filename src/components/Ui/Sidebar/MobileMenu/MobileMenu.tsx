import React, { useState } from "react";
import { Drawer, Fab, List, ListItem, ListItemText, useMediaQuery } from "@mui/material";
import AddIcon from "@mui/icons-material/Add";
import { useNavigate } from "react-router-dom";
import styles from "../Sidebar.module.css";

interface MobileMenuProps {
    categories: any[];
    open: boolean;
    onClose: () => void;
}

const MobileMenu: React.FC<MobileMenuProps> = ({ categories, open, onClose }) => {
    const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
    const navigate = useNavigate();

    const handleCategoryClick = (categoryName: string) => {
        navigate(`/${categoryName}`);
        onClose();
        setSelectedCategory(categoryName);
    };

    return (
        <Drawer anchor="top" open={open} onClose={onClose}  sx={{ height: "100%"}}>
            <div
                style={{
                    display: "flex",
                    flexDirection: "column",
                    alignItems: "center",
                    justifyContent: "center",
                    height: "100%",
                }}
            >
            <img
                src="/images/logo.png"
                alt="Росатом"
                className={styles.logoImage}
                style={{ margin: "1em 1em" }}
            />
            <List sx={{ flexDirection: "column", alignItems: "center", paddingTop: "5px", width: "90%" }}>
                {categories.map((item) => (
                    <ListItem
                        key={item.resource_id}
                        onClick={() => handleCategoryClick(item.name)}
                        sx={{
                            backgroundColor: selectedCategory === item.name ? "#e0e0e0" : "transparent",
                            borderRadius: "7px",
                        }}
                    >
                        <ListItemText primary={item.name} />
                    </ListItem>
                ))}
            </List>
            <Fab
                variant="extended"
                sx={{
                    width: "40%",
                    backgroundColor: "#035FA2",
                    color: "white",
                    marginTop: "2em",
                    marginBottom: "2em",
                    display: "flex",
                    justifyContent: "space-between",
                    transition: "background-color 0.5s",
                    fontSize: "15px",
                    "&:hover": {
                        backgroundColor: "#024978",
                    },
                }}
                size="large"
            >
                Документ
                <AddIcon />
            </Fab>
            </div>
        </Drawer>
    );
};

export default MobileMenu;
