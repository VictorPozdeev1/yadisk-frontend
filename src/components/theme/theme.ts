import { createTheme, responsiveFontSizes } from '@mui/material/styles';
import { Theme } from '@mui/system';

// переопределяем инерфейс брейкпоинтов
declare module '@mui/material/styles' {
  interface BreakpointOverrides {
    xs: false; // removes the `xs` breakpoint
    sm: false;
    md: false;
    lg: false;
    xl: false;
    mobile: true; // adds the `mobile` breakpoint
    tablet: true;
    laptop: true;
    desktop: true;
  }
}
const themeColors={
  fontDark:"#181818",
  actionBlue:'#035FA2',
  secondary:'#EDEDED',

}
export const appTheme:Theme = createTheme({
  typography: {
    fontFamily: 'Manrope, Roboto, Arial',

    h1:{
      fontWeight:'bold',
    },
    h2:{//category title
      fontSize:36,
      lineHeight:1.5,
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h3:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h4:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h5:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h6:{//doc table header
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    body1:{
      color:themeColors.fontDark,
      fontWeight:'600',
    },
  },
  breakpoints:{
    values:{
      mobile:320,
      tablet:805,
      laptop:1200,
      desktop:1523,
    }
  },

  palette: {
    primary: {
      main: themeColors.actionBlue,//action-blue
      light: themeColors.secondary,
      contrastText: themeColors.fontDark,//font-dark
    },
    secondary:{//for table buttons
      main: themeColors.fontDark,//font-dark
    }
  },

});
