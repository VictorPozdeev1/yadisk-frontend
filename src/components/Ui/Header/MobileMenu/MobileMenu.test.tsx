import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from "@mui/material";
import { appTheme } from '../../../theme/theme';
import MobileMenu from './MobileMenu';
import userEvent from '@testing-library/user-event';

const onClose = jest.fn()

describe('MobileMenu component', () => {
    test('Render component', () => {
        render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <MobileMenu
                        open={true}
                        onClose={onClose}
                    />
                </ThemeProvider>
            </BrowserRouter>
        )

        expect(screen.getByText('Все документы')).toBeInTheDocument();
        expect(screen.getByRole('link')).toBeInTheDocument();
        
        userEvent.click(screen.getByRole('button'))
        expect(onClose).toBeCalled();

        userEvent.click(screen.getByTestId('drawer'))
        expect(onClose).toBeCalled();

        userEvent.click(screen.getByTestId('navlink'))
        expect(onClose).toBeCalled();

    })

    test('MobileMenu snapshot', () => {
        const mobileMenu = render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <MobileMenu
                        open={true}
                        onClose={onClose}
                    />
                </ThemeProvider>
            </BrowserRouter>
        )
        expect(mobileMenu).toMatchSnapshot();
    })
})