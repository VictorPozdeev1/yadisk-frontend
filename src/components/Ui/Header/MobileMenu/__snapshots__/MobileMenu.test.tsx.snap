// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu component MobileMenu snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-10iz68-MuiModal-root-MuiDrawer-root"
      data-testid="drawer"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorRight css-1160xiw-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          data-testid="toggle_el"
          style="display: flex; flex-direction: column; align-items: center; justify-content: start; height: 100vh; width: 320px;"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-n9xjd5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-1k04ev7-MuiSvgIcon-root"
              data-testid="MenuOpenIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h13v-2H3v2zm0-5h10v-2H3v2zm0-7v2h13V6H3zm18 9.59L17.42 12 21 8.41 19.59 7l-5 5 5 5L21 15.59z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            style="width: 90%; background-color: rgb(237, 237, 237); border-radius: 7px; text-align: center;"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
              style="padding: 12px 24px 12px 24px; display: block; width: 100%; text-decoration: none; color: rgb(3, 95, 162); cursor: pointer;"
            >
              Все документы
            </a>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
